### PR TITLE
Fix (FP-4348): Fixed the left menu for account

### PR DESCRIPTION
**Pull request by KenwayTheGreat**

### CODE REVIEW

**Comment by frontastic-developers** [comment]
## :white_check_mark: Tests [2023.06.01.11.51-test](https://thegoodstore.frontastic.io/developer/continuousIntegration?build=2023.06.01.11.51-test&branch=fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages) on branch [`fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages`](https://github.com/frontastic-developers/customer-thegoodstore/tree/fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages) successful

Commit [`e1836264`](https://github.com/frontastic-developers/customer-thegoodstore/commit/e18362642d23e7581d5e9756f2e9a1098c93ae0e) by *KenwayTheGreat* in 1:53

File: packages/home/frontend/components/commercetools-ui/organisms/account/index.tsx [hunk 1]
```diff
@@ -74,26 +74,44 @@ const AccountDetails: React.FC<AccountDetailsProps> = ({
...
+    return {
+      'edit-personal-info': <PersonalInfoForm />,
+      'edit-newsletter': <SubscribeForm />,
+      'change-password': <ChangePasswordForm />,
+      'delete-account': <DeleteAccountForm />,
+    };
+  }, []);
+  const account = useMemo(
+    () =>
+      accountPagesRef[id as 'edit-personal-info' | 'edit-newsletter' | 'change-password' | 'delete-account'] ?? (
```
Comment (abdotaker608) [inline packages/home/frontend/components/commercetools-ui/organisms/account/index.tsx:87]: Let's not do this hard coding, whenever we add an entry to this `accountPagesRef` you need to update this as well. You can use `as keyof typeof accountPagesRef` instead

File: packages/home/frontend/components/commercetools-ui/organisms/account/index.tsx [hunk 2]
```diff
@@ -74,26 +74,44 @@ const AccountDetails: React.FC<AccountDetailsProps> = ({
...
+        <MyAccount isLoading={isLoading} />
+      ),
+    [id, isLoading, accountPagesRef],
+  );
+
+  const addressPagesRef = useMemo(() => {
+    return { add: <AddressForm />, edit: <AddressForm editedAddressId={id?.split('_')[1]} /> };
+  }, [id]);
+  const addresses = useMemo(() => {
+    return addressPagesRef[id?.split('-')[0] as 'add' | 'edit'] ?? <Addresses />;
```
Comment (abdotaker608) [inline packages/home/frontend/components/commercetools-ui/organisms/account/index.tsx:97]: Same as above `as keyof typeof addressPagesRef`

**Review by abdotaker608 — CHANGES_REQUESTED** [review]

**Comment by frontastic-developers** [comment]
## :white_check_mark: Tests [2023.06.05.10.59-test](https://thegoodstore.frontastic.io/developer/continuousIntegration?build=2023.06.05.10.59-test&branch=fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages) on branch [`fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages`](https://github.com/frontastic-developers/customer-thegoodstore/tree/fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages) successful

Commit [`cb07b5b8`](https://github.com/frontastic-developers/customer-thegoodstore/commit/cb07b5b8025cc6cca7dd6506555eb1bebfb6f66f) by *KenwayTheGreat* in 1:52

**Comment by frontastic-developers** [comment]
## :white_check_mark: Tests [2023.06.05.12.49-test](https://thegoodstore.frontastic.io/developer/continuousIntegration?build=2023.06.05.12.49-test&branch=fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages) on branch [`fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages`](https://github.com/frontastic-developers/customer-thegoodstore/tree/fix/FP-4348-category-in-left-navigation-dropdown-should-stay-active-when-opening-sub-pages) successful

Commit [`e874fd1b`](https://github.com/frontastic-developers/customer-thegoodstore/commit/e874fd1b09b64c618fa2acef7252bd18a19abd21) by *KenwayTheGreat* in 2:05

**Review by abdotaker608 — APPROVED** [review]

**Review by xcintas — APPROVED** [review]

**Pull request merged**